packages/flutter/lib/src/widgets/app.dart

```diff
@@ -269,16 +267,16 @@ class WidgetsApp extends StatefulWidget {
 
   /// Creates a [WidgetsApp] that uses the [Router] instead of a [Navigator].
   WidgetsApp.router({
-    Key key,
+    Key? key,
     this.routeInformationProvider,
-    @required this.routeInformationParser,
-    @required this.routerDelegate,
-    BackButtonDispatcher backButtonDispatcher,
+    required this.routeInformationParser,
+    required this.routerDelegate,
```

Comment:
these two should have non-nullable types in the parameter list as well as being required

Reply:
Done